x-pack/plugins/security/server/routes/authentication/common.ts

```diff
@@ -51,18 +51,28 @@ export function defineCommonRoutes({ router, authc, basePath, logger }: RouteDef
         } catch (error) {
           return response.customError(wrapIntoCustomErrorResponse(error));
         }
-      })
+      }
     );
   }
 
-  router.get(
-    { path: '/internal/security/me', validate: false },
-    createLicensedRouteHandler(async (context, request, response) => {
-      try {
-        return response.ok({ body: (await authc.getCurrentUser(request)) as any });
-      } catch (error) {
-        return response.customError(wrapIntoCustomErrorResponse(error));
-      }
-    })
-  );
+  // Generate two identical routes with new and deprecated URL and issue a warning if route with deprecated URL is ever used.
+  for (const path of ['/internal/security/me', '/api/security/v1/me']) {
+    router.get(
+      { path, validate: false },
+      createLicensedRouteHandler(async (context, request, response) => {
+        if (path === '/api/security/v1/me') {
+          logger.warn(
```

Comment:
__note:__ Assuming we make this route internal I don't mention the name of the internal route as we don't provide any guarantees regarding it.